waitsensor/waitsensor.go

```diff
@@ -121,48 +128,48 @@ func NewThresholds(t map[string]int) []Bin {
 }
 
 type BoundingBox struct {
-	relBox []float64
+	xMin, yMin, xMax, yMax float64
 }
 
-func NewBoundingBox(coords []float64) (BoundingBox, error) {
-	if len(coords) != 0 {
-		if len(coords) != 4 {
-			return BoundingBox{}, errors.Errorf("bounding box must contain 4 numbers [x_min, y_min, x_max, y_max], got %v.", coords)
-		}
-		for _, e := range coords {
-			if e < 0.0 || e > 1.0 {
-				return BoundingBox{}, errors.New("bounding box numbers are relative to the image dimension, and must be numbers between 0 and 1.")
-			}
-		}
-		xMin, yMin, xMax, yMax := coords[0], coords[1], coords[2], coords[3]
-		if xMin >= xMax {
-			return BoundingBox{}, fmt.Errorf("x_min (%f) must be less than x_max (%f)", xMin, xMax)
-		}
-		if yMin >= yMax {
-			return BoundingBox{}, fmt.Errorf("y_min (%f) must be less than y_max (%f)", yMin, yMax)
+func NewBoundingBox(coords BoundingBoxConfig) (BoundingBox, error) {
+	coordList := []float64{coords.XMin, coords.YMin, coords.XMax, coords.YMax}
+	for _, e := range coordList {
+		if e < 0.0 || e > 1.0 {
+			return BoundingBox{}, errors.New("bounding box numbers are relative to the image dimension, and must be numbers between 0 and 1.")
 		}
 	}
-	return BoundingBox{relBox: coords}, nil
+	if coords.XMin >= coords.XMax {
+		return BoundingBox{}, fmt.Errorf(
+			"x_min (%f) must be less than x_max (%f)", coords.XMin, coords.XMax)
+	}
+	if coords.YMin >= coords.YMax {
+		return BoundingBox{}, fmt.Errorf(
+			"y_min (%f) must be less than y_max (%f)", coords.YMin, coords.YMax)
+	}
+	bb := BoundingBox{xMin: coords.XMin,
+	                  yMin: coords.YMin,
+	                  xMax: coords.XMax,
+	                  yMax: coords.YMax,
+	                  }
+	return bb, nil
 }
 
 // crop coordinates were already validated upon configuration
 // empty bounding box implies no crop
 func (bb BoundingBox) Crop(img image.Image) image.Image {
-	coords := bb.relBox
-	if len(coords) != 4 {
+	if bb.xMax == 0 || bb.yMax == 0 {
```

Comment:
I'm open to other definitions of "invalid bounding boxes should return the whole image," if you want something else.

Reply:
I'm all right with that